### PR TITLE
Automated script execution and collection of information.

### PR DESCRIPTION
**Pull request by Sima214**
Try to define a protocol between the scripts and the gui by creating a `Registry`.

~~`Registry` is a class, whose objects come with an `update()` method which (re)loads the enumaration from the scripts. The enumeration is stored in the field `root` of `Registry` objects which acts like a dictionary of BaseEntry elements.~~

~~Currently I am having trouble with how the python scripts should get access to the *Entry classes. For now storing references of them in a dict and passing that to the python scripts seems to work, but it's not the 'cleanest' method either.~~

Protocol documentation in scripts/Protocol.md

### CODE REVIEW

**Comment by Sima214** [comment]
Cleaner and more robust registry implementation.
Did not update the documentation.
For now see scripts/examples/*ex.* for usage.

**Comment by Sima214** [comment]
Ready to be accepted and merged.
Please read `scripts/README.md` for more information about the operation of this patch.

File: scripts/cpu/name.sh [hunk 1]
```diff
@@ -1,4 +0,0 @@
-#!/bin/bash
-
```
Comment (MaanooAk) [inline scripts/cpu/name.sh]: Can you leave those two scripts as examples, we can remove them later, I can modify the output to match the protocol if you don't want to

File: genume.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env python3
+
```
[the remark this answers is not in the document]
Comment (Sima214) [reply genume.py]: I am still trying to understand how packages work in python. I will try a better solution...

File: genume/globals.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from pathlib import Path
+
+# A string representing our current version.
+VERSION = "0.0.0"
```
[the remark this answers is not in the document]
Comment (Sima214) [reply genume/globals.py]: I am pretty sure you can change it on my branch. Also it's literally a placeholder.

File: scripts/cpu/name.sh (hunk 1 above)
Comment (Sima214) [reply scripts/cpu/name.sh]: I was planning a more sophisticated cpu support as a follow-up pull request...

**Pull request merged**